test_runner/performance/test_perf_ingest_using_pgcopydb.py

```diff
@@ -0,0 +1,343 @@
+        # Remove 'ms' part from the string to prevent double-counting
+        duration_str = duration_str.replace(ms_match.group(0), "")
+
+    # Convert and add hours
+    h_match = hours_pattern.search(duration_str)
+    if h_match:
+        hours = int(h_match.group(1))
+        total_seconds += hours * 3600
+
+    # Convert and add minutes
+    m_match = minutes_pattern.search(duration_str)
+    if m_match:
+        minutes = int(m_match.group(1))
+        total_seconds += minutes * 60
+
+    # Convert and add seconds
+    s_match = seconds_pattern.search(duration_str)
+    if s_match:
+        seconds = int(s_match.group(1))
+        total_seconds += seconds
+
+    return total_seconds
+
+
+def parse_log_and_report_metrics(log_file_path: Path, backpressure_time_diff: float):
+    """Parses the pgcopydb log file for performance metrics and reports them to the database."""
+    metrics = {"backpressure_time": backpressure_time_diff}
+
+    # Define regex patterns to capture metrics
+    metric_patterns = {
+        "COPY_INDEX_CONSTRAINTS_VACUUM": re.compile(
+            r"COPY, INDEX, CONSTRAINTS, VACUUM \(wall clock\).*"
+        ),
+        "COPY_CUMULATIVE": re.compile(r"COPY \(cumulative\).*"),
+        "CREATE_INDEX_CUMULATIVE": re.compile(r"CREATE INDEX \(cumulative\).*"),
+        "CONSTRAINTS_CUMULATIVE": re.compile(r"CONSTRAINTS \(cumulative\).*"),
+        "FINALIZE_SCHEMA": re.compile(r"Finalize Schema.*"),
+        "TOTAL_DURATION": re.compile(r"Total Wall Clock Duration.*"),
+    }
+
+    # Parse log file
+    with log_file_path.open("r") as log_file:
+        for line in log_file:
+            for metric_name, pattern in metric_patterns.items():
+                if pattern.search(line):
+                    # Extract duration and convert it to seconds
+                    duration_match = re.search(r"\d+h\d+m|\d+s|\d+ms|\d+\.\d+s", line)
+                    if duration_match:
+                        duration_str = duration_match.group(0)
+                        duration_seconds = convert_to_seconds(duration_str)
+                        metrics[metric_name] = duration_seconds
+
+    # Report metrics to the database
+    report_metrics_to_db(metrics)
+
+
+def get_endpoint_id():
+    """Extracts and returns the first segment of the hostname from the PostgreSQL URI stored in BENCHMARK_INGEST_TARGET_CONNSTR."""
+    connstr = os.getenv("BENCHMARK_INGEST_TARGET_CONNSTR")
+    if connstr is None:
+        raise OSError("BENCHMARK_INGEST_TARGET_CONNSTR environment variable is not set.")
+
+    # Parse the URI
+    parsed_url = urlparse(connstr)
+
+    # Extract the hostname and split to get the first segment
+    hostname = parsed_url.hostname
+    if hostname is None:
+        raise ValueError("Unable to parse hostname from BENCHMARK_INGEST_TARGET_CONNSTR")
+
+    # Split the hostname by dots and take the first segment
+    endpoint_id = hostname.split(".")[0]
+
+    return endpoint_id
+
+
+def report_metrics_to_db(metrics):
```

Comment:
Can we use [`zenbenchmark`](https://github.com/neondatabase/neon/blob/38563de7dd2aa910c4d5564a4ad8c67ab62334e3/test_runner/fixtures/benchmark_fixture.py#L450-L476) fixture instead? (that is what we usually use for collection and uploading results)